### ISSUE TITLE
[bug] Option names with hyphen character is not deleted correctly

### ISSUE THREAD

**Issue by uilianries** [conan-io/conan#10570]
That's a limitation with python which creates an inconsistent situation with Conan recipes.

When naming an option using hyphen e.g. `foo-bar`, on python, it will be parsed as `foo_bar`, because `-` means minus operator.

    del self.options.foo-bar    # SyntaxError: cannot delete operator
    del self.options.foo_bar   # valid

However, once deleted with underscore character, the option is still valid when passing option as command argument.

    conan create . foo/0.1@ -o foo:foo-bar=True  # works

The current workaround is using `delattr`.

    delattr(self.options, "foo-bar")

We can't restrict option names to underscore only, but we need to remove accordingly and show the no option error message.

Related to https://github.com/conan-io/conan-center-index/pull/9248

### Environment Details (include every applicable attribute)
  * Operating System+version: Any
  * Compiler+version: Any
  * Conan version: 1.45.0
  * Python version: 3.5+

### Steps to reproduce (Include if Applicable)

```
from conans import ConanFile
class FooConan(ConanFile):
    options = {"foo-bar": [True, False]}
    default_options = {"foo-bar": False}

    def config_options(self):
        del self.options.foo_bar
````

```
conan create . foo/0.1@ -o foo:foo_bar=False
ERROR: foo/0.1: option 'foo_bar' doesn't exist
Possible options are ['foo-bar']

conan create . foo/0.1@ -o foo:foo-bar=False
Works normally ...
```

### Expected Behavior

The follow statement must return an error:

    conan create . foo/0.1@ -o foo:foo-bar=False
    ERROR: foo/0.1: option 'foo-bar' doesn't exist

**Comment by memsharded** [comment]
I think there is a misunderstanding about what is happening: ``del self.options.foo_bar   # valid`` is not removing anything. The ``del`` operator is coded so it doesn't fail when the option doesn't exist that is all (and can't be changed without breaking). Given the option name, I see no other possibility than doing it via ``delattr``, we are not converting hyphens to underscores, that is also problematic at least if not fully breaking.

**Comment by uilianries** [comment]
> (and can't be changed without breaking)

I was afraid you would say that  :joy: 

Can we at least, improve it for Conan 2.0?

**Comment by memsharded** [comment]
> Can we at least, improve it for Conan 2.0?

How? 😅 
You mean, by raising an error for ``del self.options.foo_bar`` if it doesn't exist? I think that has already been addressed in 2.0, but I can check, yes.

Update: Yes, we are good:

```
ERROR: pkg/0.1: Error in config_options() method, line 28
        del self.options.some_thing
        ConanException: option 'some_thing' doesn't exist
Possible options are ['shared']
```

**Comment by uilianries** [comment]
Amazing, at least we will no longer suffer it again.

We are good to close this issue. Thank you for clarifying.
